### PR TITLE
add some debugging doc, after i found out how to print debug messages

### PR DESCRIPTION
**Pull request by pvinis**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ This allows you to use the system and audio control key codes.
 
 `CONSOLE_ENABLE`
 
-TODO
+This allows you to print messages that can be read using `hid_listen`. Add this to your `Makefile`, and set it to `yes`. Then put `println`, `printf`, etc. in your keymap or anywhere in the `qmk` source. Finally, open `hid_listen` and enjoy looking at your printed messages.
```
Comment (exiva) [inline readme.md]: Might want to add a link with where to get the `hid_listen` tool.

**Comment by jackhumbert** [comment]
Nice - thanks!

**Comment by fredizzimo** [comment]
I was going to comment on this before it was merged, but never had time to do it.

I think that the 
```
dprint
dprintf
dprintln
dmsg
```
Functions should be documented too. 

The output of those are are controlled by the debug enable magic and debug_config, and it can further be refined to include/exclude matrix, keyboard and/or mouse debug output.

These could be used for keymaps as well. Theoretically the unused bits of the debug config could be used too, and you could even bind keys to modify that, and it could be saved in the eeprom. But there's no guarantee that it will stay so in the future, so it's probably not a very good idea. 

I'm not sure how and where it should be documented and how much detail would be needed though.

**Comment by pvinis** [comment]
that sounds useful. well, i didnt know what they can do, so i only used 'print'. all the more reason to document them in some way..

**Pull request merged**